connectors-e2e-test/connectors-e2e-test-rabbitmq/src/test/java/io/camunda/connector/e2e/RabbitMqTests.java

```diff
@@ -0,0 +1,183 @@
+            .property("routing.exchange", EXCHANGE_NAME)
+            .property("routing.routingKey", ROUTING_KEY)
+            .property("routing.virtualHost", VIRTUAL_HOST)
+            .property("routing.hostName", HOST_NAME)
+            .property("routing.port", PORT)
+            .property("message.body", MESSAGE)
+            .property("resultVariable", "result")
+            .writeTo(new File(tempDir, "template.json"));
+
+    ZeebeTest bpmnTest = setupTestWithBpmnModel("rabbitMqTask", elementTemplate);
+
+    RabbitMqResult result = RabbitMqResult.success();
+    assertThat(bpmnTest.getProcessInstanceEvent()).hasVariableWithValue("result", result);
+
+    String receivedMessage = consumeMessage();
+
+    if (receivedMessage == null) {
+      fail("No message received within the expected timeout.");
+    } else {
+      assertEquals(MESSAGE, receivedMessage, "The sent message should match the received message");
+    }
+  }
+
+  @Test
+  public void uriAuthenticationSendMessageTest() throws Exception {
+    String uri =
+        String.format(
+            "amqp://%s:%s@%s:%d/%s",
+            rabbitMQContainer.getAdminUsername(),
+            rabbitMQContainer.getAdminPassword(),
+            rabbitMQContainer.getHost(),
+            rabbitMQContainer.getAmqpPort(),
+            "%2F");
+
+    var elementTemplate =
+        ElementTemplate.from(ELEMENT_TEMPLATE_PATH)
+            .property("authentication.authType", "uri")
+            .property("authentication.uri", uri)
+            .property("routing.exchange", EXCHANGE_NAME)
+            .property("routing.routingKey", ROUTING_KEY)
+            .property("message.body", MESSAGE)
+            .property("resultVariable", "result")
+            .writeTo(new File(tempDir, "template.json"));
+
+    ZeebeTest bpmnTest = setupTestWithBpmnModel("rabbitMqTask", elementTemplate);
+
+    RabbitMqResult result = RabbitMqResult.success();
+    assertThat(bpmnTest.getProcessInstanceEvent()).hasVariableWithValue("result", result);
+
+    String receivedMessage = consumeMessage();
+    if (receivedMessage == null) {
+      fail("No message received within the expected timeout.");
+    } else {
+      assertEquals(MESSAGE, receivedMessage, "The sent message should match the received message");
+    }
+  }
+
+  private String consumeMessage() throws Exception {
+    String receivedMessage;
+    try (Connection connection = factory.newConnection();
+        Channel channel = connection.createChannel()) {
+      BlockingQueue<String> blockingQueue = new ArrayBlockingQueue<>(1);
+      DeliverCallback deliverCallback =
+          (consumerTag, delivery) -> {
+            String devMessage = new String(delivery.getBody(), StandardCharsets.UTF_8);
+            blockingQueue.add(devMessage);
+          };
+      String consumerTag =
+          channel.basicConsume(QUEUE_NAME, true, deliverCallback, consumerTag1 -> {});
+      receivedMessage = blockingQueue.poll(1, TimeUnit.SECONDS);
```

Comment:
What about using:
```
GetResponse basicGet(String  queue, boolean autoAck)
```
 instead of the more complex callback based approach?

https://javadoc.io/doc/com.rabbitmq/amqp-client/5.7.3/com/rabbitmq/client/Channel.html#basicGet-java.lang.String-boolean-

Reply:
Great, thanks!